backbone.js

```diff
@@ -1447,16 +1457,12 @@
         name = '';
       }
       if (!callback) callback = this[name];
-      var router = this;
-      Backbone.history.route(route, function(fragment) {
-        var args = router._extractParameters(route, fragment);
-        if (router.execute(callback, args, name) !== false) {
-          router.trigger.apply(router, ['route:' + name].concat(args));
-          router.trigger('route', name, args);
-          Backbone.history.trigger('route', router, name, args);
-        }
-      });
-      return this;
+      var handler = {
+        route: route,
+        callback: callback,
+        name: name
+      };
+      Router.handlers.push(handler);
```

Comment:
I had that in my first revision, but I changed it to mimic the current behavior to be less breaking. The current behavior is that if routerA and routerB are both using the same instance of history, and both register the same route, then only one of those handlers will be called, and never both. This is likely why the handlers array is currently stored on the history instance.